front-end/src/components/analysis-page/IntentMenu.test.js

```diff
@@ -120,6 +123,151 @@ describe("IntentMenu", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for Intent 0 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 1 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+
+        it ("should display two intents when two intents are both starred and flagged", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            props.intents[1].star = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+
+        it ("should display two intents when two intents are both starred and flagged and menu is closed and opened", () => {
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            props.intents[1].star = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // Close intent menu
+            userEvent.click(screen.getByLabelText("Close intent menu"));
+            // Button amount stays the same even when closed
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+            // To get make the filters false for future testing
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+        });
+    });
```

Comment:
I haven't done a full CR so I might be wrong on a couple details of implementation, but here are my thoughts:

Maybe? What IntentMenu should be tested for is whether it can properly display 0, 1, 2, ..., n intents in the list. It doesn't care about how those are filtered, and those filters are tested in the IntentSearch tests. However, if you scroll up you can see that I also wrote similar tests for whether 0, 1, 2, etc. intents display when we type into the search bar. We need at least 1 of those tests to check the proper formatting (the part of the intent contents that matches the search should be highlighted), but the idea behind having multiple tests is to make sure that the search string is correctly being passed to IntentSearch. So same idea here - these tests are testing whether the filter buttons in IntentMenu are actually changing the parameters in IntentSearch, which can't be tested for in the latter's test file. Question is how necessary it is - I don't think there's too much harm in keeping them.

Reply:
👍then sounds good. Wasn't sure if my intuition was correct. Glad I asked for a second opinion

Reply:
Actually I had a second thought on this and now agree with Sam's comment - I will add a separate comment for this.